src/NextGen/Gateways/Stripe/NextGenStripeGateway/NextGenStripeRepository.php

```diff
@@ -0,0 +1,147 @@
+use Give\PaymentGateways\Gateways\Stripe\Actions\SaveDonationSummary;
+use Give\PaymentGateways\Stripe\ApplicationFee;
+use Stripe\Customer;
+use Stripe\Exception\ApiErrorException;
+use Stripe\PaymentIntent;
+
+trait NextGenStripeRepository {
+    /**
+     * @unreleased
+     * @throws ApiErrorException
+     */
+    protected function generateStripePaymentIntent($accountId, Money $amount): PaymentIntent
+    {
+        return PaymentIntent::create(
+            [
+                'amount' => $amount->formatToMinorAmount(),
+                'currency' => $amount->getCurrency()->getCode(),
+                'automatic_payment_methods' => ['enabled' => true],
+            ],
+            ['stripe_account' => $accountId]
+        );
+    }
+
+    /**
+     * Get or create Stripe Customer from Donation
+     *
+     * @unreleased
+     * @throws ApiErrorException
+     * @throws Exception
+     */
+    public function getOrCreateStripeCustomerFromDonation(
+        string $connectAccountId,
+        Donation $donation
+    ): Customer {
+        $donorCustomerId = give_stripe_get_customer_id($donation->email) ?? '';
+
+        // make sure customerId still exists in  connect account
+        if ($donorCustomerId) {
+            $customer = Customer::retrieve($donorCustomerId, ['stripe_account' => $connectAccountId]);
+        }
+
+        // create a new customer if necessary
+        if (!$donorCustomerId || !$customer) {
+            $customer = Customer::create(
+                [
+                    'name' => "$donation->firstName $donation->lastName",
+                    'email' => $donation->email,
+                ],
+                ['stripe_account' => $connectAccountId]
+            );
+        }
+
+         DonationNote::create([
+            'donationId' => $donation->id,
+            'content' => sprintf(__('Stripe Customer ID: %s', 'give'), $customer->id)
+        ]);
+
+        if ($customer->id !== $donorCustomerId) {
+            give()->donor_meta->update_meta($donation->donorId, give_stripe_get_customer_key(), $customer->id);
+        }
+
+        give_update_meta($donation->id, give_stripe_get_customer_key(), $customer->id);
+
+        return $customer;
+    }
+
+     /**
+     * @unreleased
+     * @throws ApiErrorException
+     */
+    protected function updateStripePaymentIntent(string $id, array $data): PaymentIntent
+    {
+        return PaymentIntent::update(
+            $id,
+            $data
+        );
+    }
+
+    /**
+     * @unreleased
+     *
+     * @throws InvalidPropertyName
+     */
+    protected function getPaymentIntentArgsFromDonation(Donation $donation, Customer $customer)
+   {
+       // Collect intent args to be updated
+        $intentArgs = [
+            'amount' => $donation->amount->formatToMinorAmount(),
+            'customer' => $customer->id,
+            'description' => (Call::invoke(SaveDonationSummary::class, $donation))->getSummaryWithDonor(),
```

Comment:
Now that we're on PHP 7.0, I don't think we actually need to use `Call::invoke` anymore. It will also track itself better if we don't in the IDE.

Reply:
yes your right! c53b0225